src/tokenserver/auth/oauth.rs

```diff
@@ -0,0 +1,113 @@
+use actix_web::{web, Error};
+use async_trait::async_trait;
+use pyo3::{
+    prelude::{Py, PyAny, PyErr, PyModule, Python},
+    types::{IntoPyDict, PyString},
+};
+use serde::{Deserialize, Serialize};
+use serde_json;
+use tokio::time;
+
+use super::VerifyToken;
+use crate::tokenserver::{error::TokenserverError, settings::Settings};
+
+use core::time::Duration;
+use std::convert::TryFrom;
+
+/// The information extracted from a valid OAuth token.
+#[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct VerifyOutput {
+    #[serde(rename = "user")]
+    pub fxa_uid: String,
+    pub generation: Option<i64>,
+}
+
+/// The verifier used to verify OAuth tokens.
+#[derive(Clone)]
+pub struct RemoteVerifier {
+    // Note that we do not need to use an Arc here, since Py is already a reference-counted
+    // pointer
+    inner: Py<PyAny>,
```

Comment:
If `Py` didn't use an Arc internally, we'd need to use one here to maintain keep-alive connections between threads

Reply:
https://docs.rs/pyo3/0.15.1/pyo3/struct.Py.html#impl-Clone